tests/unit/Endpoints/Webhooks/ProductHandlerTest.php

```diff
@@ -65,69 +49,211 @@ public function testCreateProduct()
+
+        $galleryImages = $woocommerceUpdatedProduct->get_gallery_image_ids();
+        $productImages = $updatedProductData->get('flat_product.product_images');
+        foreach ($productImages as $index => $images) {
+            if ($images['id'] === $updatedProductData->get('flat_product.main_image.id')) {
+                unset($productImages[$index]);
+            }
+        }
+        $this->assertNotSameSize($galleryImages, $productImages, 'Product images should not be updated');
+
+        // Regular price should not be updated
+        if (self::WC_TYPE_CONFIGURABLE !== $woocommerceUpdatedProduct->get_type()) {
+            $expectedRegularPrice = $updatedProductData->get('product_default_price.ppu_wt');
+            $this->assertNotEquals(
+                $expectedRegularPrice,
+                $woocommerceUpdatedProduct->get_regular_price(),
+                "[sku=$sku] WooCommerce regular price should not match expected regular price"
+            );
+
+            // Discounted price should not be updated
+            if ($updatedProductData->get('product_price.ppu_wt') !== $expectedRegularPrice) {
+                $expected_discounted_price = $updatedProductData->get('product_price.ppu_wt');
+                $this->assertNotEquals(
+                    $expected_discounted_price,
+                    $woocommerceUpdatedProduct->get_sale_price(),
+                    "[sku=$sku] WooCommerce discount price should not match the expected discount price"
+                );
+            }
+        }
+
+        // No cross-sell products should be set
+        $crossSellIds = $woocommerceUpdatedProduct->get_cross_sell_ids();
+        $this->assertEmpty($crossSellIds, 'Cross-sell products should not be set');
+
+        // No up-sell products should be set
+        $upSellIds = $woocommerceUpdatedProduct->get_upsell_ids();
+        $this->assertEmpty($upSellIds, 'Up-sell products should not be set');
+
+        $this->assertProductStock($updatedProductData, $woocommerceUpdatedProduct, $sku);
+    }
+
+    public function testUpdateProductPricesOnly()
+    {
+        $this->mockCreateProductRequestWithTest();
+        $woocommerceUpdatedProduct = $this->mockUpdateProductRequestWithTest(self::UPDATE_PRICES_DATADUMP_HOOK);
+
+        // Get the updated data dump as a dotnotated collection
+        $updatedFile = $this->getDataDump(self::UPDATE_DATADUMP_DIRECTORY.'/'.self::UPDATE_DATADUMP_PRODUCT);
+        $updatedProductData = $updatedFile->getReturn()['data'];
+        $updatedProductData = new Dot($updatedProductData[0]);
+
+        $this->assertTaskNotCount(0, 'There are not supposed to be any tasks left to process');
+
+        // Other details like SKU and images should not be updated
+        $sku = $woocommerceUpdatedProduct->get_sku();
+        $this->assertNotEquals($updatedProductData['sku'], $sku, 'Product\'s SKU should not be updated');
```

Comment:
```suggestion
        $this->assertNotEquals($updatedProductData['sku'], $sku, 'Product\'s SKU should not be updated');
        $this->assertEquals($productBeforeHook['sku'], $sku, 'Product\'s SKU should not be updated');
```
It would make sense also to check if it was same a before, it could have been changed to some random string which will make this test pass.

This all applies to other `$this->assertNotEquals` in the `public function testUpdate***Only()` tests.

Reply:
Added positive testing for all `not`.